tools/LogCollector/source/update_mgmt_health_check.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/python
+
+'''
+   OMS Health Check Script for Update Management Linux VM's
+
+   Authors, Reviewers & Contributors :
+                 Tommy Nguyen Azure Automation SE
+                 Atchut Barli Azure Automation Sr. SE
+                 Shipra Malik Ohri Azure Automation Sr. SE
+                 Shujun Liu Azure Automation Sr. SE
+                 Adrian Doyle Sr. Esc. Eng.
+                 Brian McDermott Sr. Esc. Eng.
+
+   Date        : 2018-07-06
+   Version     : 1.0
+   
+'''
+
+import sys
+import os
+import os.path
+import socket
+import datetime
+
+from os import walk
+
+def main(output_path=None):
+
+    output = []
+
+    output.append(check_network() + "\n")
+    output.append(check_oms_agent_installed() + "\n")
+    output.append(check_oms_agent_running() + "\n")
+    output.append(check_oms_admin() + "\n")
+    output.append(check_hybrid_worker_running() + "\n")
+    output.append(get_machine_info() + "\n")
+
+    output.append("\n")
+
+    if output_path is not None:
+        try: 
+            os.makedirs(output_path)
+        except OSError:
+            if not os.path.isdir(output_path):
+                raise
+        f = open(output_path + "/healthcheck-" + str(datetime.datetime.utcnow().isoformat()) + ".log", "w")
+        f.write("".join(output))
+
+    print "".join(output)
+
+def check_network():
+    output = []
+    output.append("Network check: \n\n")
+
+    endpoints = ["bing.com", "google.com"]
+    
+    agent_regions = ["jpe", "eus2", "cc", "scus", "uks", "sea"]
+    agent_endpoint = "-agentservice-prod-arm-1.trafficmanager.net"
+
+    jrds_regions = ["wcus", "ncus"]
```

Comment:
JRDS is available in more regions , the following blog has the list from a year back https://docs.microsoft.com/en-us/azure/log-analytics/log-analytics-oms-gateway  but please check the service xmls.

ideally, if the script can find that from conf file or where ever hybrid worker is picking from , it will need less maintenance from our end to update periodically.

Reply:
worker.conf will have the jrds info